### ISSUE TITLE
Document error population

### ISSUE THREAD

**Issue by crichards17** [crichards17/Distributed_Id_Allocator#24]
Document which methods on the wasm-id-allocator may populate the error_string field.

**Comment by crichards17** [comment]
@taylorsw what kind of documentation were you looking for here?

**Comment by taylorsw** [comment]
For hot-path methods (generate and normalize), I suggest we doc what NAN means and how to get errors. For the others, I think we may be able to really simplify things by returning a Result with a JS error: https://docs.rs/wasm-bindgen/latest/wasm_bindgen/struct.JsError.html

This would actually cause an exception to be thrown automatically, which would be really nice for the TS-side of things I think. That likely should be its own issue, but probably a precursor to this one.

**Comment by taylorsw** [comment]
@crichards17 specifically, set_cluster_capacity, get_token, finalize_range, decompress, recompress, and deserialize could all return Result<SOMETHING, JsError> and remove any usage of set_error and simplify code in general.

**Comment by crichards17** [comment]
Makes sense. I'll make the other issue for returning JSError. For the purpose of those hot paths where we'll retain the set_error, I'm wondering mechanically how you want that documentation to be integrated (method-level comment documentation? Top-level doc?) -- system documentation is new to me.

**Comment by taylorsw04** [comment]
Docs on those methods. I don't think it'll get generated on the JS side (although it would be neat if it is).

**Comment by taylorsw04** [comment]
Oh hey, it's my work account.

**Comment by taylorsw04** [comment]
I have doxxed myself.
